forest/data.py

```diff
@@ -407,15 +407,31 @@ def load_image(path, variable, itime, ipressure):
     return load_image_pts(path, variable, (itime,), (itime, ipressure))
 
 
+def _image_cache(hash_func):
+    """Simple wrapper to cache images per-server"""
+    def inner(load_func):
+        def innermost(*args):
+            key = hash_func(*args)
+            if key not in IMAGES:
+                # TODO: Replace this infinite cache
+                IMAGES[key] = load_func(*args)
+            return IMAGES[key]
+        return innermost
+    return inner
+
+
+def _image_hash(path, variable, pts_3d, pts_4d):
+    """Convert arguments to hashable type"""
+    return (path, variable, pts_hash(pts_3d), pts_hash(pts_4d))
+
+
+@_image_cache(_image_hash)
```

Comment:
Could you use https://docs.python.org/3/library/functools.html#functools.lru_cache instead?

Reply:
LRU is tempting but in this case we want to cache function calls with unhashable args and we can easily calculate the size of an image, e.g. sum(array.nbytes for array in ...). So a total size custom-hash lru cache is what I'm shooting for, but could go down the simpler functools route if my refactoring proves fruitless

Reply:
Admittedly, LRU is perfect if all images are the same size (e.g. 256x256 float32 arrays) which we'll migrate to in the near future

Reply:
Hmm, might be looking in the wrong place. Start-up memory is biggish (~180MB) and turning caching off still accretes memory, albeit at a much slower pace (~0.1MB per click)

Reply:
Difficult to get reliable data with gc cycles